js/app/widgets/dynamicLogo.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016 Endless Mobile, Inc.
+
+/* exported DynamicLogo */
+
+const EosKnowledgePrivate = imports.gi.EosKnowledgePrivate;
+const Gtk = imports.gi.Gtk;
+const Gio = imports.gi.Gio;
+const GObject = imports.gi.GObject;
+
+const Knowledge = imports.app.knowledge;
+const ThemeableImage = imports.app.widgets.themeableImage;
+const FormattableLabel =imports.app.widgets.formattableLabel;
+const Utils = imports.app.utils;
+
+/**
+ * Class: DynamicLogo
+ *
+ * This widget implemenets our logo system. A logo can be composed of a
+ * symbol, a wordmark or both. The way these elements are arranged is
+ * defined by css custom properties.
+ *
+ * -EknDynamicLogo-layout:
+ *  * 'vertical' - symbol and wordmark are always aligned vertically. Default.
+ *  * 'horizontal' - symbol and wordmark are always aligned horizontally.
+ *  * 'auto' - symbol and wordmark alignment depends on constraints (not implemented).
+ *
+ * -EknDynamicLogo-mode:
+ *  * 'full' - both symbol and wordmark are visible. Default.
+ *  * 'image' - only the image is visible.
+ *  * 'text' - only the text is visible.
+ *  * 'auto' - visibility depends on constraints (not implemented).
+ */
+const DynamicLogo = new Knowledge.Class({
+    Name: 'DynamicLogo',
+    Extends: Gtk.Box,
+
+    StyleProperties: {
+        'layout': GObject.ParamSpec.string('layout', 'layout', 'layout',
```

Comment:
Hard to tell if these are more like style properties or regular properties. Do you expect them to be style-able in response to a widget state change?